### PR TITLE
Add `IF EXISTS` to `ALTER USER`

### PR DESCRIPTION
**Pull request by Hunterness**
Depends on https://github.com/neo-technology/neo4j/pull/9014 (merged)

### CODE REVIEW

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 1]
```diff
@@ -288,7 +288,7 @@ The following commands are only executable against the `system` database:
...
 | <<administration-security-users-alter-password, ALTER CURRENT USER SET PASSWORD FROM ... TO>>                                                                                     | User and role  | Change the password of the user that is currently logged in.
-| <<administration-security-users-alter, ALTER USER ... [SET [PLAINTEXT \| ENCRYPTED] PASSWORD {password [CHANGE [NOT] REQUIRED] \| CHANGE [NOT] REQUIRED}] [SET STATUS {ACTIVE \| SUSPENDED}]>>             | User and role  | Changes a user account. Changes can include setting a new password, setting the account status and enabling that the user should change the password upon next login.
+| <<administration-security-users-alter, ALTER USER ... [IF EXISTS] [SET [PLAINTEXT \| ENCRYPTED] PASSWORD {password [CHANGE [NOT] REQUIRED] \| CHANGE [NOT] REQUIRED}] [SET STATUS {ACTIVE \| SUSPENDED}]>>             | User and role  | Changes a user account. Changes can include setting a new password, setting the account status and enabling that the user should change the password upon next login.
```
Comment (OliviaYtterbrink) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc:291]: Still think this looks horrible and should be fixed in general, but that's a different discussion...

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/UserManagementTest.scala [hunk 2]
```diff
@@ -54,6 +54,14 @@ Set a new password for a user. This user will not be required to change this pas
 ###assertion=update-one parameters=update
 //
 
+ALTER USER alice IF EXISTS SET PASSWORD CHANGE REQUIRED
+###
+
+Required to change the password on the next login for a user, if the user exists.
```
Comment (OliviaYtterbrink) [inline cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/UserManagementTest.scala]: This sentence feels a bit weird... 🤔

**Review by OliviaYtterbrink — CHANGES_REQUESTED** [review]

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc (hunk 1 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/glossary.asciidoc:291]: I agree on that

File: cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/UserManagementTest.scala (hunk 2 above)
Comment (Hunterness) [reply cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/UserManagementTest.scala]: I know, I did not come up with anything that sounded good, if you have any idea, please help :P

Comment (mnd999) [reply cypher/refcard-tests/src/test/scala/org/neo4j/cypher/docgen/refcard/UserManagementTest.scala]:
```suggestion
If the specified user exists, force this user to change their password on the next login.
```

**Review by OliviaYtterbrink — APPROVED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala [hunk 3]
```diff
@@ -183,6 +183,13 @@ class SecurityUserAndRoleManagementTest extends DocumentingTest with QueryStatis
         query("SHOW USERS", assertAllNodesShown("User", column = "user")) {
           resultTable()
         }
+        p("This command is optionally idempotent, with the default behavior to throw an exception if the user does not exists. " +
```
Comment (martin-neotech) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala]:
```
p("""This command is optionally idempotent, with the default behavior to throw an exception if the user does not exists.
     #Appending `IF EXISTS` to the command will ensure that no exception is thrown and nothing happens should the user not exist.""".stripMargin('#'))
```

**Review by martin-neotech — COMMENTED** [review]
Minor asciidoc style.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala [hunk 4]
```diff
@@ -183,6 +183,13 @@ class SecurityUserAndRoleManagementTest extends DocumentingTest with QueryStatis
...
+          "Appending `IF EXISTS` to the command will ensure that no exception is thrown and nothing happens should the user not exist.")
```
Comment (Hunterness) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SecurityUserAndRoleManagementTest.scala]:
```suggestion
        p("""This command is optionally idempotent, with the default behavior to throw an exception if the user does not exists.
             #Appending `IF EXISTS` to the command will ensure that no exception is thrown and nothing happens should the user not exist.""".stripMargin('#'))
```

**Review by martin-neotech — APPROVED** [review]

**Pull request merged**